platforms/software/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -470,6 +469,7 @@ public void testLogging(Action<? super TestLoggingContainer> action) {
 
     @TaskAction
     public void executeTests() {
+        filter.validate();
```

Comment:
❓ Why do we need to call this here now?

Reply:
Before we called `validate()` in a setter, e.g.:
```
    @Override
    public TestFilter setExcludePatterns(String... testNamePatterns) {
        // this does also validation
        return setFilteringPatterns(excludeTestNames, testNamePatterns);
    }
```

Now we have `SetProperty` and we cannot do validation when `property = value` is set. So we call validation before the task runs

Reply:
Looks like we are missing a validation feature for providers...